### ISSUE TITLE
Slim: Data Lost or Disorderly When Reading Data from tfrecord Files

### ISSUE THREAD

**Issue by arlcurten** [tensorflow/models#5129]
### System information
- **What is the top-level directory of the model you are using**: Research/Slim
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
 Yes, we tested the problem with few parameters changed, a tfPrint() to display data, and some tfrecord file with batch of same lable (please see my situation below)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Bazel version (if compiling from source)**: 0.15.0
- **CUDA/cuDNN version**: None (CPU used)
- **GPU model and memory**: None (CPU used)
- **Exact command to reproduce**: use fine-tuning_inception-v3_on_flowers shell script to run the example but need to generate tfrecord file with serial tags

### Describe the problem
Our datasets are from videos so we hope they can be read in their original order. We have 200 videos and we extracted 32 frames from each of them, tagging them with the same tag number for the same video. After disable the shuffle function, we still found that the order is changed or some data are missing as below when doing training. We used `tf.Print()` to print out the tags after `tf.train.batch()`. It appeared like:
`labels=[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0]
....
labels=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1]
....
labels=[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2]
....`
It didn't happen in each data read, just randomly decrease a few tags sometimes. When we applied the same method to evaluation, it was totally fine to show the result we want, like:
`labels=[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1]
labels=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
labels=[2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2]`
A bunch of 32 tags was read together, no data missing or disorderly.
I am not sure if it is a bug or not? Also, how can we make sure our data is read in order and without missing?
Thanks

### Source code / logs
in `train_image_classifier.py`
##############################################################
Create a dataset provider that loads data from the dataset
##############################################################
`with tf.device(deploy_config.inputs_device()):
  provider = slim.dataset_data_provider.DatasetDataProvider(
      dataset,
      num_readers=FLAGS.num_readers,
      shuffle=_data_shuffle,
      common_queue_capacity=20 * FLAGS.batch_size,
      common_queue_min=10 * FLAGS.batch_size)
  [image, label] = provider.get(['image', 'label'])
  label -= FLAGS.labels_offset
  train_image_size = FLAGS.train_image_size or network_fn.default_image_size

  image = image_preprocessing_fn(image, train_image_size, train_image_size)

  images, labels = tf.train.batch(
      [image, label],
      batch_size=FLAGS.batch_size,
      num_threads=FLAGS.num_preprocessing_threads,
      capacity=5 * FLAGS.batch_size)

  labels = tf.Print(labels,[labels], "....labels=",summarize=32)

  labels = slim.one_hot_encoding(
      labels, dataset.num_classes - FLAGS.labels_offset)
  batch_queue = slim.prefetch_queue.prefetch_queue(
      [images, labels], capacity=2 * deploy_config.num_clones) `

Only `tf.Print()` is added in it.

We added `shuffle=false` and set `num_readers=1` in the script, trying to let the data_provider() read data in sequence without parallel jobs. For tf.train.batch(), `num_preprocessing_threads=1` is also set in script.

**Comment by arlcurten** [comment]
Hi, I am still working on the project. Is there anything I can try first and report to here?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by arlcurten** [comment]
Hi,
The project is not on-going anymore. I will close it then.
